dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api.h

```diff
@@ -654,6 +660,27 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
 
   ThreadPool thread_pool_;
   static constexpr int kOutputDim = 3;
+
+ private:
+  void UpdateTestCounters(int nsamples_hw, int nsamples_cuda, int nsamples_host) {
+    nsamples_hw_ += nsamples_hw;
+    nsamples_cuda_ += nsamples_cuda;
+    nsamples_host_ += nsamples_host;
+  }
+
+
+  /**
+   * Registers counters, used only for unit-test reasons.
+   */
+  void RegisterTestCounters() {
+    RegisterDiagnostic("nsamples_hw", &nsamples_hw_);
+    RegisterDiagnostic("nsamples_cuda", &nsamples_cuda_);
+    RegisterDiagnostic("nsamples_host", &nsamples_host_);
+    RegisterDiagnostic("using_hw_decoder", &using_hw_decoder_);
+  }
+
+  // HW/CUDA Utilization test counters
+  long long int nsamples_hw_ = 0, nsamples_cuda_ = 0, nsamples_host_ = 0;  // NOLINT
```

Comment:
I went for `long long` since it's at least 64 bit, so in case some broader type appears, it will be handled. The NOLINT because linter doesn't like `long long`

Reply:
I would stick to int64_t. 2^64 samples is a lot.

Reply:
Done